### PR TITLE
[frontend]  Add Attack pattern list in Knowledge/Attack pattern with massive operations (#4230)

### LINKED ISSUE #4230: Add Attack pattern list in Knowledge/Attack pattern with massive operations
## Use case

Being able to use massive operation on attack pattern list associated with an SDO

## Current Workaround

None

## Proposed Solution

Add a simple list view of the attack pattern in the Knowledge/attack pattern tab, like in other tabs.

### PR DESCRIPTION
**Pull request by Gwendoline-FAVRE-FELIX**
## Proposed changes

On Matrix View:

- Add a new online view of the matrix with massive operations
- Moving the matrix filter
- Remove "Display only used techniques"

On Kill chain View anc Courses of actions View:

- Add the matrix name before the category title
- Remove “Display only used  techniques” and “Enable invert colors” (hide it)

## Related issues

- https://github.com/OpenCTI-Platform/opencti/issues/4230

### CODE REVIEW

**Comment by Archidoit** [comment]
In an entity>Knowledge>AttackPattern>Matrix view : there is only one kill chain in the kill chain list, even if other kill chains exist : 
image

**Comment by Archidoit** [comment]
When hovering the 'enable inverted colors' icon, the circle is not centered on the icon : 
image

**Comment by Archidoit** [comment]
In the Matrix in-line view: add a filter on name with for instance : name don't start with XXX
You have entities with name starting with XX ! (see screenshort)
It's because it's not really the name that is displayed in the 'name' column, but the representative...
Either display only the name in the 'name' column (like in the Attack pattern screen), or change the column name for 'representative'.

image

**Comment by Gwendoline-FAVRE-FELIX** [comment]
But I've got all the Kill chains in the filter.
Kill Chain Filter

**Comment by Gwendoline-FAVRE-FELIX** [comment]
I've corrected the hover and the name column. As for the display in the kill chain filter, you need to relaunch your frontend for it to be up to date.

**Comment by Archidoit** [comment]
In Matrix in line view :
- Columns sorting not working
- when clicking on a label chip in the list, the associated label filter is not added
- the 'add filtering' option is not present for columns even if we can filter on them
image

**Comment by Archidoit** [comment]
The already added attack patterns are not checked in the 'create relationship' edition form list, even after reload : 
image


Maybe try to fix this in this PR?

**Comment by Archidoit** [comment]
When you click on export to image or pdf from the matrix in line view : 
image


the screen is blocked here : 
image

**Comment by Archidoit** [comment]
inconsistency with elsewhere in the platform: the filters are placed on the same line than the search bar 
image


(instead of just below) : 
image

**Comment by Archidoit** [comment]
Exports don't take filters into account

- Add a filter in the attack pattern matrix in line list
image

- Do an export
- The whole list of attack pattern is exported (instead of only the attack patterns matching the filter) :

**Comment by Archidoit** [comment]
Filters are repercuiting elsewhere:

- Add a filter in the attack patterns matrix in line list
image

- Reload the page
- The added filter is present in every 'create a relationship' lists
image

**Comment by Archidoit** [comment]
- missing view buttons tooltip when we are in Matrix in line view
image

- need to reload to see added attack patterns in matrix in line view

File: opencti-platform/opencti-front/lang/front/fr.json [hunk 1]
```diff
@@ -1425,6 +1425,7 @@
   "Kill": "Tuer",
   "Kill all": "Tuer tout",
   "Kill chain": "Kill chain",
+  "Kill chain :": "Chaîne de mise à mort :",
```
Comment (aHenryJard) [inline opencti-platform/opencti-front/lang/front/fr.json]:
```suggestion
  "Kill chain :": "Kill chain :",
```

File: opencti-platform/opencti-front/lang/front/es.json [hunk 2]
```diff
@@ -1425,6 +1425,7 @@
   "Kill": "Matar",
   "Kill all": "Matar a todos",
   "Kill chain": "«Kill chain»",
+  "Kill chain :": "Cadena de muerte :",
```
Comment (aHenryJard) [inline opencti-platform/opencti-front/lang/front/es.json]:
```suggestion
  "Kill chain :": "«Kill chain» :",
```

**Comment by Archidoit** [comment]
Two different typo for the tooltip 'matrix in line view' depending on which view we are : 
image


image

--> chose with or without '-' and do the appropriated translations

File: opencti-platform/opencti-front/lang/front/en.json [hunk 3]
```diff
@@ -1625,6 +1626,8 @@
   "Markings": "Markings",
   "Match": "Match",
   "Matches all indicator main observable types if none is listed.": "Matches all indicator main observable types if none is listed.",
+  "Matrix in line view": "Matrix in line view",
```
Comment (Archidoit) [inline opencti-platform/opencti-front/lang/front/en.json:1629]: useless

File: opencti-platform/opencti-front/src/private/components/common/stix_domain_objects/StixDomainObjectAttackPatternsKillChain.tsx [hunk 4]
```diff
@@ -284,7 +284,7 @@ const StixDomainObjectAttackPatternsKillChain: FunctionComponent<StixDomainObjec
                 />
               </ToggleButton>
             </Tooltip>
-            <Tooltip title={t_i18n('Matrix in-line view')}>
+            <Tooltip title={t_i18n('Matrix in line view')}>
```
Comment (Archidoit) [inline opencti-platform/opencti-front/src/private/components/common/stix_domain_objects/StixDomainObjectAttackPatternsKillChain.tsx:287]: To do : 
Remove 'Matrix in-line view' translations
Add 'Matrix in line view' translations

**Review by Archidoit — APPROVED** [review]

**Pull request merged**